src/app/partials/books-table/books-table.component.html

```diff
@@ -1,24 +1,71 @@
-<table mdbTable class="table">
+<table *ngIf="allBooks.length; else noneBooks" mdbTable class="table" hover="true">
   <thead class="white-text header">
   <tr>
-    <th *ngFor="let head of headElements" scope="col">{{head}} </th>
+    <th *ngFor="let head of headers; let i = index"
+        scope="col"
+        [mdbTableSort]="allBooks"
+        [sortBy]="headers[i].name">
+      {{head.name}}
+      <mdb-icon *ngIf="head.isSort"
+                icon="sort"
+                class="sort-icon">
+      </mdb-icon>
+    </th>
   </tr>
   </thead>
   <tbody class="body">
-  <tr mdbTableCol *ngFor="let book of allBooks">
-    <td class="td image">
+  <tr mdbTableCol *ngFor="let book of allBooks; let i = index">
+    <td *ngIf="i+1 >= firstItemIndex && i < lastItemIndex" class="td image">
       <img src="{{book.image}}" class="picture">
     </td>
-    <td class="td">{{book.title}}</td>
-    <td class="td">{{book.author}}</td>
-    <td class="td description">{{book.description}}</td>
-    <td class="td count">{{book.availableCount}}/{{book.count}}</td>
-    <td class="td labels">
+    <td *ngIf="i+1 >= firstItemIndex && i < lastItemIndex" class="td">
+      {{book.title}}
+    </td>
+    <td *ngIf="i+1 >= firstItemIndex && i < lastItemIndex" class="td">
+      {{book.author}}
+    </td>
+    <td *ngIf="i+1 >= firstItemIndex && i < lastItemIndex" class="td description">
+      {{book.description}}
+    </td>
+    <td *ngIf="i+1 >= firstItemIndex && i < lastItemIndex" class="td count">
+      {{book.availableCount}}/{{book.count}}
+    </td>
+    <td *ngIf="i+1 >= firstItemIndex && i < lastItemIndex" class="td labels">
       <a href="" *ngFor="let label of getBookLabels(book)">
-        <mdb-badge primary="true">{{label.name}}</mdb-badge>
+        <mdb-badge class="label-book">
+          {{label.name}}
+        </mdb-badge>
       </a>
     </td>
+    <td *ngIf="i+1 >= firstItemIndex && i < lastItemIndex" class="td event">
```

Comment:
move condition "i+1 >= firstItemIndex && i < lastItemIndex" to .ts setter ' isShowBook(index)'
firstItemIndex and lastItemIndex to private.